custom-configs/src/main/java/com/sequenceiq/cloudbreak/service/CustomConfigsService.java

```diff
@@ -0,0 +1,225 @@
+    }
+
+    public CustomConfigs clone(NameOrCrn nameOrCrn, String newName, String accountId) {
+        return nameOrCrn.hasName() ? cloneByName(nameOrCrn.getName(), newName, accountId) : cloneByCrn(nameOrCrn.getCrn(), newName, accountId);
+    }
+
+    public CustomConfigs cloneByName(String name, String newCustomConfigsName, String accountId) {
+        Optional<CustomConfigs> customConfigsByName = Optional.of(customConfigsRepository.findCustomConfigsByNameAndAccountId(name, accountId)
+                .orElseThrow(() -> new NotFoundException("Custom configs with name " + name + " does not exist. Cannot be cloned.")));
+        CustomConfigs newCustomConfigs = new CustomConfigs(customConfigsByName.get());
+        Set<CustomConfigProperty> newConfigSet = Set.copyOf(newCustomConfigs.getConfigs());
+        // removing associations of new configs set with original configs set
+        newConfigSet.forEach(config -> {
+            config.setId(null);
+            config.setCustomConfigs(null);
+        });
+        newCustomConfigs.setConfigs(newConfigSet);
+        newCustomConfigs.setName(newCustomConfigsName);
+        return create(newCustomConfigs, accountId);
+    }
+
+    public CustomConfigs cloneByCrn(String crn, String newCustomConfigsName, String accountId) {
+        Optional<CustomConfigs> customConfigsByCrn = Optional.of(customConfigsRepository.findCustomConfigsByResourceCrn(crn))
+                .orElseThrow(() -> new NotFoundException("Custom configs with crn " + crn + " not found. Cannot be cloned."));
+        CustomConfigs newCustomConfigs = new CustomConfigs(customConfigsByCrn.get());
+        Set<CustomConfigProperty> newConfigSet = Set.copyOf(customConfigsByCrn.get().getConfigs());
+        newConfigSet.forEach(config -> {
+            config.setId(null);
+            config.setCustomConfigs(null);
+        });
+        newCustomConfigs.setConfigs(newConfigSet);
+        newCustomConfigs.setName(newCustomConfigsName);
+        return create(newCustomConfigs, accountId);
+    }
+
+    public CustomConfigs deleteByCrn(String crn) {
+        Optional<CustomConfigs> customConfigsByCrn = Optional.of(customConfigsRepository.findCustomConfigsByResourceCrn(crn)
+                .orElseThrow(NotFoundException.notFound("CustomConfigs", crn)));
+        // since we're outside core module and core module has dependency on custom configs,
+        // Adding core module as a dependency to this one would cause circular dependency.
+        customConfigsRepository.deleteById(customConfigsByCrn.get().getId());
+        return customConfigsByCrn.get();
+    }
+
+    public CustomConfigs deleteByName(String name, String accountId) {
+        Optional<CustomConfigs> customConfigsByName = Optional.of(customConfigsRepository.findCustomConfigsByNameAndAccountId(name, accountId)
+                .orElseThrow(NotFoundException.notFound("CustomConfigs", name)));
+        customConfigsRepository.deleteById(customConfigsByName.get().getId());
```

Comment:
create a method getByName and use this everywhere.

Reply:
Updated.